students/nam_vo/lesson09/mailroom_oo/donor_models.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+
+"""
+Class responsible for donor data encapsulation.
+"""
+class Donor(object):
+    def __init__(self, name, donations=[0]):
+        # Name must be a string
+        if isinstance(name, str):
+            self._name = name
+        else:
+            raise TypeError("name must be a string!")
+        # All donations must be non-negative number (integer or float)
+        for donation in donations:
+            try:
+                if donation < 0:
+                    raise ValueError("donation can't be negative")
+            except TypeError:
+                print("donation must be a non-negative number")
+        # Convert all donations to floats
+        self._donations = [float(donation) for donation in donations]
+           
+    @property
+    def name(self):
+        return self._name
+    
+    @name.setter
+    def name(self, value):
+        # Name must be a string
+        if isinstance(value, str):
+            self._name = value
+        else:
+            raise TypeError("name must be a string!")
+
+    @property
+    def donations(self):
+        return self._donations
+
+    @donations.setter
+    def donations(self, value):
+        # All donations must be non-negative number (integer or float)
+        for donation in value:
+            try:
+                if donation < 0:
+                    raise ValueError("donation can't be negative")
+            except TypeError:
+                print("donation must be a non-negative number")
+        # Convert all donations to floats
+        self._donations = [float(donation) for donation in value]
+
+    def __str__(self):
+        return '{}, {}'.format(self.name, self.donations)
+
+    def send_thankyou(self):
+        """Compose an email thanking the donor for their generous donation."""
+        content = "\n\n".join([
+            f"\nDear {self.name},",
+            "Thank you for your generous donation in the amount of ${:.2f}! Have a great day!\n".format(self.donations[len(self.donations) - 1]),
+        ])
+
+        return content
+
+    def add_donation(self, amount):
+        """Add new amount to the donation list"""
+        self.donations += [amount]
+
+    @property
+    def sum_donations(self):
+        return sum(self.donations)
+        
+    @staticmethod
+    def sort_key(self):
+        return self.sum_donations
+
+"""
+Class responsible for donor collection data encapsulation.
+"""
+class DonorCollection(Donor):
```

Comment:
DonorCollection does not inherit from Donor - this is an incorrect relationship.

DonorCollect should hold a reference to multiple donor objects.